### ISSUE TITLE
Creating a model and comparing the accuracy of different algorithms

### ISSUE THREAD

**Issue by Suvodeep-Das** [VishalSinghParmar2001/Relating-Social-Media-to-Stock-Movement-Public#4]
Full Name: Suvodeep Das
GitHub Profile Link: https://github.com/Suvodeep-Das
Objective: Creating a model to and testing accuracy using different algorithms(Random Forest, Logistic Regression, Decision Tree).

I would like to work on this issue.

**Comment by Suvodeep-Das** [comment]
GSSoC'22 participant

**Comment by VishalSinghParmar2001** [comment]
I am reviewing get back to you asap!!

**Comment by Suvodeep-Das** [comment]
Sure

**Comment by mar-galho** [comment]
Hi @Suvodeep-Das. It will be great to have you working on this issue.

**Comment by Suvodeep-Das** [comment]
Can you please provide me with the CSV file?

**Comment by mar-galho** [comment]
Yes. The files are here. Also, if you want, you can upload them here - feel free to create another issue to handle this task

Link: https://drive.google.com/drive/folders/18DmYKhI_-zUo6ZyNdKnoPSMSDYXW-JuG?usp=sharing

**Comment by visheshks04** [comment]
Hey @ricardoprins. I have a few suggestions to improve the model. 

1. The class imbalance of the model is keeping it from learning the pattern. The model at the moment is very much overfitted towards one class. I can improve this by balancing the classes by upsampling the other class so the support balances out for both the classes.
2. The current logistic regression model may not be the best algorithm. I can try better algorithms like Naive Bayes or time series models like LSTM.
3. I will also try to perform some hyperparameter tuning which has not been done yet.

**Comment by visheshks04** [comment]
Please let me know if this can be assigned to me.

**Comment by mar-galho** [comment]
This surely can, your suggestions seem very sound. 

@Suvodeep-Das and @visheshks04 - how can we improve efficiency here? should this current issue/work be split into two (or more) different tasks, each corresponding to a step of the process and/or different algorithms?

**Comment by visheshks04** [comment]
@ricardoprins First things first. I think one checkpoint can be balancing the classes which has to be done before any training. I am happy to do that as well as some training following that.

If you wish to split the issue. My suggestion is this. One split can be balancing the classes. The other can be using some conventional model along with hyperparameter tuning. Another can be using some sequence model along with hyperparameter tuning.
What do you say?

**Comment by mar-galho** [comment]
I agree with your assessment. @Suvodeep-Das , what do you say? Shall we go ahead and split the issue into three different ones? If you don't object in the next 24h, I'll do it.

As for the class balancing task, let's go ahead and have that issue created. I'll assign that to you, @visheshks04 . 

Thanks for your valuable input!

**Comment by Suvodeep-Das** [comment]
@ricardoprins Yes it will be great if you split the issues - firstly balanced fitting the model followed hyperparameter tuning and finding the best algorithm for the model.

**Comment by mar-galho** [comment]
Alright, could you go ahead and create the issues? I appreciate your assistance!

**Comment by Suvodeep-Das** [comment]
Ya sure!
